src/api/app/controllers/webui/users/bs_requests_controller.rb

```diff
@@ -42,25 +42,48 @@ def set_bs_requests
                                      .or(@bs_requests.where(bs_request_actions: { source_package_id: User.session.involved_packages }))
-                       when @selected_filter['involvement'].include?('review')
+                       when ['review']
                          @bs_requests.where(reviews: { user: User.session })
                                      .or(@bs_requests.where(reviews: { group: User.session.groups }))
                                      .or(@bs_requests.where(reviews: { project: User.session.involved_projects }))
                                      .or(@bs_requests.where(reviews: { package: User.session.involved_packages }))
+                       when %w[incoming outgoing]
+                         @bs_requests.where(creator: User.session)
+                                     .or(@bs_requests.where(bs_request_actions: { target_project_id: User.session.involved_projects }))
+                                     .or(@bs_requests.where(bs_request_actions: { target_package_id: User.session.involved_packages }))
+                                     .or(@bs_requests.where(bs_request_actions: { source_project_id: User.session.involved_projects }))
+                                     .or(@bs_requests.where(bs_request_actions: { source_package_id: User.session.involved_packages }))
+                       when %w[incoming review]
+                         @bs_requests.where(bs_request_actions: { target_project_id: User.session.involved_projects })
+                                     .or(@bs_requests.where(bs_request_actions: { target_package_id: User.session.involved_packages }))
+                                     .or(@bs_requests.where(reviews: { user: User.session }))
+                                     .or(@bs_requests.where(reviews: { group: User.session.groups }))
+                                     .or(@bs_requests.where(reviews: { project: User.session.involved_projects }))
+                                     .or(@bs_requests.where(reviews: { package: User.session.involved_packages }))
+                       when %w[outgoing review]
+                         @bs_requests.where(creator: User.session)
+                                     .or(@bs_requests.where(bs_request_actions: { source_project_id: User.session.involved_projects }))
+                                     .or(@bs_requests.where(bs_request_actions: { source_package_id: User.session.involved_packages }))
+                                     .or(@bs_requests.where(reviews: { user: User.session }))
+                                     .or(@bs_requests.where(reviews: { group: User.session.groups }))
+                                     .or(@bs_requests.where(reviews: { project: User.session.involved_projects }))
+                                     .or(@bs_requests.where(reviews: { package: User.session.involved_packages }))
+                       else
+                         @bs_requests
```

Comment:
Maybe this big switch case can be rewritten like [this one](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/concerns/webui/notifications_filter.rb#L7)? When I wrote that method I found it very easy to read and understand especially thanks to the `inject.(:or)`

Reply:
Let's try :)

Reply:
done

Reply:
Great! It looks smoothed, thanks.